nginx/tests/with_mock.bats

```diff
@@ -0,0 +1,54 @@
+setup() {
+	load 'test_helper/bats-support/load' # this is required by bats-assert
+	load 'test_helper/bats-assert/load'
+
+	set -eu -o pipefail
+}
+
+[[ -n "${DEBUG_BATS:-}" ]] && set -x
+
+@test "API should respond before timeout" {
+	run timeout 15 bash -c "until curl -fksm5 https://localhost:1443; do sleep 0.1; done"
+	assert_success
+}
+
+@test "HTTP request should redirect to HTTPS" {
+	run curl -fsm5 -I http://localhost:1080 
+	assert_success
+	assert_line --partial --index 0 'HTTP/1.1 301 Moved Permanently'
+	assert_line --partial 'Location: https://localhost/'
+}
+
+@test "HTTP acme-challenge should not redirect" {
+	run curl -Ism5 http://localhost:1080/.well-known/acme-challenge/
+	assert_success
+ 	refute_line --index 0 'HTTP/1.1 301 Moved Permanently'
+	assert_line --partial --index 0 'HTTP/1.1 404 Not Found'
+}
+
+@test "should recieve response from CHT api" {
+	run curl -fksm5 https://localhost:1443/
+	assert_success
+	assert_output 'Hello from CHT api mock'
+
+	run curl -fksm5 https://localhost:1443/somepath
+	assert_success
+	assert_output 'Test'
+
+	run curl -Ifksm5 https://localhost:1443/doesnotexist
+	assert_failure
+	assert_line --partial --index 0 'HTTP/1.1 404 Not Found'
```

Comment:
Option 1
```suggestion
	run curl -Ifkm5 https://localhost:1443/doesnotexist
	assert_failure
	assert_line --partial 'HTTP/1.1 404 Not Found'
```
Option 2
```suggestion
	run curl -Iksm5 https://localhost:1443/doesnotexist
	assert_line --partial --index 0 'HTTP/1.1 404 Not Found'
```
both
```suggestion
	run curl -Ikm5 https://localhost:1443/doesnotexist
	assert_line --partial 'HTTP/1.1 404 Not Found'
```
@mrjones-plip  this is the alternatives I was talking about

Reply:
Option 1:

```
 ✓ API should respond before timeout
 ✓ HTTP request should redirect to HTTPS
 ✓ HTTP acme-challenge should not redirect
 ✗ should recieve response from CHT api
   (from function `assert_line' in file tests/test_helper/bats-assert/src/assert.bash, line 526,
    in test file tests/with_mock.bats, line 40)
     `assert_line --partial 'HTTP/1.1 404 Not Found'' failed with status 22
   
   -- no output line contains substring --
   substring : HTTP/1.1 404 Not Found
   output (4 lines):
       % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                      Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
     curl: (22) The requested URL returned error: 404 Not Found
   --
   
 ✓ Should return 502 on connection drop
 ✓ Should return json with on connection drop with accept header
```

Option 2:

```
 ✓ API should respond before timeout
 ✓ HTTP request should redirect to HTTPS
 ✓ HTTP acme-challenge should not redirect
 ✓ should recieve response from CHT api
 ✓ Should return 502 on connection drop
 ✓ Should return json with on connection drop with accept header

6 tests, 0 failures
```

option 3:

```
bats tests/with_mock.bats
 ✓ API should respond before timeout
 ✓ HTTP request should redirect to HTTPS
 ✓ HTTP acme-challenge should not redirect
 ✓ should recieve response from CHT api
 ✓ Should return 502 on connection drop
 ✓ Should return json with on connection drop with accept header

6 tests, 0 failures
```